### PR TITLE
Add doc on StashCache VO registration (SOFTWARE-3518)

### PR DESCRIPTION
**Pull request by matyasselmeci**
This is the VO part of SOFTWARE-3518; putting VO registration in the install-cache and install-origin docs would have made them too big so I split it out.

### CODE REVIEW

File: docs/data/stashcache/vo-registration.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+- `/astro/PROTECTED` for protected data
+
+Separating the public and protected data in separate directory trees is preferred for technical reasons.
+
+
+Registering data federation information
+---------------------------------------
+
+In order to declare your VO's role in the StashCache federation,
+you must add StashCache information to the VO's YAML file in the OSG Topology repository.
```
Comment (brianhlin) [inline docs/data/stashcache/vo-registration.md]:
```suggestion
you must add StashCache information to your VO's YAML file in the OSG Topology repository.
```

File: docs/data/stashcache/vo-registration.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+```
+DataFederations:
+  StashCache:
+    Namespaces:
+      <Namespaces section>
+    AllowedCaches:
+      <AllowedCaches list>
+    AllowedOrigins:
+      <AllowedOrigins list>
+```
```
Comment (brianhlin) [inline docs/data/stashcache/vo-registration.md]: I would change this to an example, rolling with the "Astro VO" example above

File: docs/data/stashcache/vo-registration.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  For information about setting up an origin, see the [installation document](/data/stashcache/install-origin/).
+
+- The VO must allow one or more StashCache Caches to cache their data.
+  The more places a VO's data can be cached in, the bigger the data transfer benefit for the VO.
+  The majority of caches across OSG will automatically cache all "public" VO data.
+  Caching "protected" VO data will often be done on a site owned by the VO.
+  For information about setting up a cache, see the [installation document](/data/stashcache/install-cache/).
+
+
+Choosing namespaces
```
Comment (brianhlin) [inline docs/data/stashcache/vo-registration.md]: This section seems to be logically separate from the registration itself. Maybe we should rename the document `Getting Your Data into StashCache` or something like that?

File: docs/data/stashcache/vo-registration.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+- If you have some protected data, then AllowedCaches is a list of _resources_ that are allowed to cache your data.
+   A resource is an entry in a `/topology/<FACILITY>/<SITE>/<RESOURCEGROUP>.yaml` file,
+   for example "CHTC_STASHCACHE_CACHE".
+
+   The following requirements must be met for the resource:
+
+   - It must have an "XRootD cache server" service
+   - It must have an AllowedVOs list that includes either your VO, "ANY", or "ANY_PUBLIC"
+   - It must have a DN attribute with the DN of its host cert
```
Comment (brianhlin) [inline docs/data/stashcache/vo-registration.md]: We should really have a topology page that people can consult to find a list of acceptable caches/origins

File: docs/data/stashcache/vo-registration.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+- The VO must allow one or more StashCache origins to host their data.
+  An origin will typically be hosted on a site owned by the VO.
+  For information about setting up an origin, see the [installation document](/data/stashcache/install-origin/).
+
+- The VO must allow one or more StashCache Caches to cache their data.
+  The more places a VO's data can be cached in, the bigger the data transfer benefit for the VO.
+  The majority of caches across OSG will automatically cache all "public" VO data.
+  Caching "protected" VO data will often be done on a site owned by the VO.
+  For information about setting up a cache, see the [installation document](/data/stashcache/install-cache/).
```
Comment (brianhlin) [inline docs/data/stashcache/vo-registration.md]: I think the last 3 bullets are unnecessary here and we should indicate in each section below that unique namespaces, caches, and origins are required.

File: docs/data/stashcache/vo-registration.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+In order for a VO to join the StashCache Federation, the following needs to be true:
+
+- The VO must already be registered in the OSG Topology.
+
+- The VO must pick one or more "namespaces" for their data.
+  A namespace is a directory tree in the federation where VO data is found.
+
+!!! note
+    Namespaces are global across the federation, so you must work with the StashCache Operations team
+    to ensure that your VO's namespaces do not collide with those of another VO.
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/data/stashcache/vo-registration.md]: What's a good next step? Send an email to help@osg with a subject like "Requesting StashCache namespace(s) for <VO>" and put the namespaces in the body?

File: docs/data/stashcache/vo-registration.md (hunk 2 above)
Comment (matyasselmeci) [reply docs/data/stashcache/vo-registration.md]: Should I then get rid of the Astro example in the namespaces section?

File: docs/data/stashcache/vo-registration.md (hunk 5 above)
Comment (matyasselmeci) [reply docs/data/stashcache/vo-registration.md]: Not sure exactly what you mean but I think my "Split out Requirements" commit should do it?

File: docs/data/stashcache/vo-registration.md (hunk 3 above)
Comment (matyasselmeci) [reply docs/data/stashcache/vo-registration.md]: `Getting Your VO's Data into StashCache` perhaps? I think VO should be in the name.

File: docs/data/stashcache/vo-registration.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
+Managing VO Registration for the StashCache Federation
+======================================================
+
+This document describes the registration steps required to manage a VO's role
+in the StashCache Data Federation.
+The StashCache Federation allows a site or regional network to cache data frequently used on the OSG,
+reducing data transfer over the wide-area network and decreasing access latency
+for the VO's jobs.
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/data/stashcache/vo-registration.md]: Fixed

File: docs/data/stashcache/vo-registration.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
+Managing VO Registration for the StashCache Federation
+======================================================
+
+This document describes the registration steps required to manage a VO's role
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/data/stashcache/vo-registration.md]: Fixed

File: docs/data/stashcache/vo-registration.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+- **Public data:** data that can be read by anyone.
+- **Protected data:** data that requires authentication to read.
+
+
+Requirements
+------------
+
+In order for a VO to join the StashCache Federation, the following needs to be true:
+
+- The VO must already be registered in the OSG Topology.
```
[the remark this answers is not in the document]
Comment (matyasselmeci) [reply docs/data/stashcache/vo-registration.md]: Fixed

File: docs/data/stashcache/vo-registration.md (hunk 4 above)
Comment (brianhlin) [reply docs/data/stashcache/vo-registration.md]: A form on topology.opensciencegrid.org where you enter your VO and it lists caches and origins that your VO could use.

Comment (matyasselmeci) [reply docs/data/stashcache/vo-registration.md]: Sounds good. https://opensciencegrid.atlassian.net/browse/SOFTWARE-3550

File: docs/data/stashcache/vo-registration.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (brianhlin) [reply docs/data/stashcache/vo-registration.md]: Yup, that sounds reasonable

**Comment by matyasselmeci** [comment]
Ready for re-review unless I'm missing something.

File: docs/data/stashcache/vo-data.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+
+Requirements
+------------
+
+In order for a Virtual Organization to join the StashCache Federation, the VO must already be registered in OSG Topology.
+See the [registration document](/common/registration/#registering-virtual-organizations).
+
+
+
+Choosing namespaces
```
Comment (brianhlin) [inline docs/data/stashcache/vo-data.md]:
```suggestion
Choosing Namespaces
```

File: docs/data/stashcache/vo-data.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+
+Putting this together, if your VO is named `Astro`, you should have:
+
+- `/astro/PUBLIC` for public data
+- `/astro/PROTECTED` for protected data
+
+Separating the public and protected data in separate directory trees is preferred for technical reasons.
+
+
+Registering data federation information
```
Comment (brianhlin) [inline docs/data/stashcache/vo-data.md]:
```suggestion
Registering Data Federation Information
```

File: docs/data/stashcache/vo-data.md [hunk 12]
```diff
@@ -0,0 +1,147 @@
+Getting Your VO's Data into StashCache
```
Comment (brianhlin) [inline docs/data/stashcache/vo-data.md]: To match the mkdocs entry:

```suggestion
Getting VO Data into StashCache
```

File: docs/data/stashcache/vo-data.md [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+---------------------------------------
+
+The VO must allow one or more StashCache origins to host their data.
+An origin will typically be hosted on a site owned by the VO.
+For information about setting up an origin, see the [installation document](/data/stashcache/install-origin/).
+
+In order to declare your VO's role in the StashCache federation,
+you must add StashCache information to your VO's YAML file in the OSG Topology repository.
+
+The data will look like the following:
```
Comment (brianhlin) [inline docs/data/stashcache/vo-data.md]:
```suggestion
For example, the full registration for the `Astro` VO may look something like the following:

```

**Pull request merged**